lib/libc/Kconfig

```diff
@@ -43,7 +43,6 @@ config NEWLIB_LIBC_SUPPORTED
 config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
-	depends on ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y") || (NATIVE_LIBRARY)
```

Comment:
`PICOLIBC_SUPPORTED` indeed must not depend on `TOOLCHAIN_HAS_PICOLIBC` since a toolchain that does not "have" Picolibc can still support Picolibc module compiled from source.

Note that Picolibc module is automatically selected when `PICOLIBC=y`:

https://github.com/zephyrproject-rtos/zephyr/blob/045a11fad19582b4434ea0dcf7b1cc01a4958767/lib/libc/picolibc/Kconfig#L6-L14

Picolibc module, which is compiled from source and intended to be portable, should be assumed to be supported on all toolchains unless one specifically sets `PICOLIBC_SUPPORTED=n`.